wordpress-mobile/WordPress-iOS
# Account Creation: Fix broken flow after signing up from the WordPress app

Fixes #21746
## Description
This PR fixes an issue where users signing up on the WordPress app land on the Reader while it should not be accessible. 

The PR also introduces logic for synchronizing calls to fetch the remote feature flags, making sure that the latest network call is always respected regardless of when the response comes. This is to avoid the following scenario:
1. Unauthenticated request to fetch feature flags made (Request A)
2. User logs in
3. Authenticated request to fetch feature flags made (Request B)
4. Response for request B returned and values updated
5. ❌  Response for request a returned and values overridden

## Testing Instructions

1. Delete both WordPress and Jetpack apps
2. Install WordPress
3. Create a new account via email
4. Background the app, do not close it
5. Confirm your new account's email on the same device
6. You should be navigated back to the WordPress app automatically
7. Tap "Done"
8. Tap "Not right now"
9. ✅ Make sure that you land on My Site with the tab bar & Jetpack features hidden 

## Regression Notes
1. Potential unintended areas of impact
Sign up flow on Jetpack

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
